### DISCUSSION TITLE
new warning re: sentry field in next.config.js

### DISCUSSION THREAD

**Discussion by RwwL** [vercel/next.js#42450] [Help] [answered]
I just upgraded Next.js from v12.1.0 to 12.3.2 and noticed when I start the dev app I now get a warning about my `sentry` setup in `next.config.js`: 

```
yarn  - Invalid next.config.js options detected: 
  - The root value has an unexpected property, sentry, which is not in the list of allowed properties (...full list)
```

I found this confusing because the setup is based on the Sentry docs' example here: https://docs.sentry.io/platforms/javascript/guides/nextjs/manual-setup/#disable-sentrywebpackplugin

We're using two of the properties mentioned there:

```
    sentry: {
      disableServerWebpackPlugin: ENV === 'development',
      disableClientWebpackPlugin: ENV === 'development',
    },
```

If it's not an allowed property, does that mean it's not working as the Sentry docs say it should?

**Comment by icyJoseph** [comment, marked as answer]
There has been a change to the expected shape of next.config.js, for example with `next-pwa`:

### Before

```js
module.exports = withPWA({
  pwa: {
    dest: "public",
    runtimeCaching,
    register: true,
    skipWaiting: true,
    buildExcludes: [/middleware-manifest.json$/],
  }
});

```

### After

```js
const withPWA = require("next-pwa")({
  dest: "public",
  runtimeCaching,
  register: true,
  skipWaiting: true,
  buildExcludes: [/middleware-manifest.json$/],
  disable: process.env.NODE_ENV === 'development'
})

module.exports = withPWA({
  reactStrictMode: false
});
```

###  Review

So the plugin now must receive its unique configuration items and that returns a function itself, `withPWA`, which wraps the `nextConfig` we want to use.


### Your solution

I searched the `@sentry/nextjs` package and found this bits:

- https://github.com/getsentry/sentry-javascript/blob/master/packages/nextjs/src/config/index.ts#L21-L23
- https://github.com/getsentry/sentry-javascript/blob/master/packages/nextjs/src/config/index.ts#L40-L44


So perhaps you just need to upgrade `@sentry/nextjs` as well?

**Reply by RwwL** [reply]
Yep, that looks like all there is to it, thanks very much!
